zone/spells.cpp

```diff
@@ -7409,3 +7544,108 @@ bool Mob::CheckWaterLoS(Mob* m)
 		zone->watermap->InLiquid(m->GetPosition())
 	);
 }
+
+bool Mob::IsImmuneToBotSpell(uint16 spell_id, Mob* caster)
```

Comment:
Why do we have bot-specific logic for stuff like this? Whether a spell is on a bot or not we should be running through generic immune checks, no ?

Reply:
Normal immune checks fire during the actual cast process and can also generate hate (if it's constantly looping through these checks they'll end up with random aggro that you can't predict).

I set all the cast checks to be done before sending the spell cast to allow for early escaping with how much spell/cast checking bots actually do.